CRM/Member/BAO/Membership.php

```diff
@@ -394,6 +394,9 @@ public static function create(&$params, &$ids, $skipRedirect = FALSE) {
         $params['line_item'],
         CRM_Utils_Array::value('contribution', $params)
       );
+      if (!empty($params['contribution'])) {
```

Comment:
I'm not sure we should be enhancing ways to add payments when creating a membership - this doesn't feel in sync with the general principal of creating a pending 'order' and then add payments

Reply:
We are already adding payments with status pending for other event/contribution backoffice. Likewise here we are just recording a pending payment. Isn't this correct?